datacollectiondemo/source/cpp/src/KaaDemo.cpp

```diff
@@ -49,7 +49,7 @@ int main()
      */
     kaaClient->start();
 
-    std::list<RecordFuture> futures;
+    std::list<std::pair<RecordFuture, std::size_t>> futurePairs;
```

Comment:
Seems that the code assumes `size_t` is equal to `Long`. That's not always true — Java's `Long` is 64-bit integer while `size_t` is unsigned and may be 32-bits long.